src/file_pipe_log/pipe.rs

```diff
@@ -175,6 +175,23 @@ impl<F: FileSystem> SinglePipe<F> {
     /// filesystem.
     fn sync_dir(&self, path_id: PathId) -> Result<()> {
         debug_assert!(!self.paths.is_empty());
+
+        // Cannot open directory in Windows, will end up with
+        // "Access Denied(OS Error 5)". We have to walk the
+        // directory to sync each file in it.
```

Comment:
This seems terrible in performance. Need a FIXME.

Reply:
Question: do we really need to sync dir, and need to do it under windows?

Reply:
Let's remove it. [Apparently](https://learn.microsoft.com/en-us/previous-versions/windows/it-pro/windows-server-2003/cc781134(v=ws.10)?redirectedfrom=MSDN#within-a-directory-entry-for-a-file) windows has the concept of directory metadata too, but they choose not to say anything about its persistence. I guess we can assume they are flushed to disk the moment file is created, like [badger](https://github.com/dgraph-io/badger/issues/699) did.